src/figuresClasses.ts

```diff
@@ -1,19 +1,92 @@
-export interface Figure {
+export type Shape = 'triangle' | 'circle' | 'rectangle';
+export type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error(`
+        The length of any side must be > 0
+      `);
+    } else if (!(b + c > a) || !(a + c > b) || !(a + b > c)) {
+      throw new Error(`
+        The longest side of a triangle must be >= than the sum of the other two
+      `);
+    } else {
+      this.a = a;
+      this.b = b;
+      this.c = c;
+    }
+  }
 
-export class Circle {
+  getArea(): number {
+    const s: number = (this.a + this.b + this.c) / 2;
+    const area: number
+      = Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
 
+    return Math.floor(area * 100) / 100;
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  public shape: Shape = 'circle';
+
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error(`
+        The radius must be > 0
+      `);
```

Comment:
I think in this case it'll be ok

```suggestion
      throw new Error(`The radius must be > 0`);
```